src/frontend/src/handler/create_table_as.rs

```diff
@@ -0,0 +1,266 @@
+
+    // Subblock to make sure PlanRef (an Rc) is dropped before `await` below.
+    let (query, query_mode, output_schema) = {
+        let context = OptimizerContext::new_with_handler_args(handler_args);
+        let (plan, query_mode, schema) = gen_batch_query_plan(&session, context.into(), stmt)?;
+
+        tracing::trace!(
+            "Generated query plan: {:?}, query_mode:{:?}",
+            plan.explain_to_string()?,
+            query_mode
+        );
+        let plan_fragmenter = BatchPlanFragmenter::new(
+            session.env().worker_node_manager_ref(),
+            session.env().catalog_reader().clone(),
+        );
+        (plan_fragmenter.split(plan)?, query_mode, schema)
+    };
+    tracing::trace!("Generated query after plan fragmenter: {:?}", &query);
+
+    let column_types = output_schema
+        .fields()
+        .iter()
+        .map(|f| f.data_type())
+        .collect_vec();
+
+    let mut row_stream = {
+        // Acquire hummock snapshot for execution.
+        // TODO: if there's no table scan, we don't need to acquire snapshot.
+        let hummock_snapshot_manager = session.env().hummock_snapshot_manager();
+        let query_id = query.query_id().clone();
+        let pinned_snapshot = hummock_snapshot_manager.acquire(&query_id).await?;
+
+        match query_mode {
+            QueryMode::Local => PgResponseStream::LocalQuery(DataChunkToRowSetAdapter::new(
+                local_execute(session.clone(), query, pinned_snapshot).await?,
+                column_types,
+                false,
+            )),
+            // Local mode do not support cancel tasks.
+            QueryMode::Distributed => {
+                PgResponseStream::DistributedQuery(DataChunkToRowSetAdapter::new(
+                    distribute_execute(session.clone(), query, pinned_snapshot).await?,
+                    column_types,
+                    false,
+                ))
+            }
+        }
+    };
+
+    let rows_count = {
+        // Get the row from the row_stream.
+        let first_row_set = row_stream
+            .next()
+            .await
+            .expect("compute node should return affected rows in output")
+            .map_err(|err| RwError::from(ErrorCode::InternalError(format!("{}", err))))?;
+        let affected_rows_str = first_row_set[0].values()[0]
+            .as_ref()
+            .expect("compute node should return affected rows in output");
+
+        Some(
+            String::from_utf8(affected_rows_str.to_vec())
+                .unwrap()
+                .parse()
+                .unwrap_or_default(),
+        )
+    };
+
+    // Implicitly flush the writes.
+    if session.config().get_implicit_flush() {
+        flush_for_write(&session, StatementType::INSERT).await?;
+    }
+
+    Ok(PgResponse::new_for_stream(
+        StatementType::INSERT,
+        rows_count,
+        row_stream,
+        vec![],
+    ))
+}
+
+fn gen_create_table_plan(
```

Comment:
We need to modify gen_create_table_plan() in create_table.rs to reuse it. We need to move it out of this function.
https://github.com/risingwavelabs/risingwave/blob/6d41b9399b6c0d81ec8854ea9cac3c65a08ba7b9/src/frontend/src/handler/create_table.rs#L206

I'm not sure this modify whether reduce readibility so I'm try to avoid modify other module.

Reply:
I think we don't need to change original method, just return `ColumnDef` in `convert_field_to_column_desc`?